pkg/adaptor/adaptor.go

```diff
@@ -24,15 +23,21 @@ func (a ErrAdaptor) Error() string {
 // Listen() listens on a pipe, processes data, and then emits it.
 // Stop() shuts down the adaptor
 type StopStartListener interface {
+	SampleConfig() string
+	Description() string
+	Connect() error
```

Comment:
in addition, the comment needs to be updated to include the new functions of the interface